monkey/monkey_island/cc/environment/server_config_generator.py

```diff
@@ -1,8 +1,21 @@
+import os
 from pathlib import Path
 
-from monkey_island.cc.server_utils.consts import DEFAULT_DEVELOP_SERVER_CONFIG_PATH
+from monkey_island.cc.environment.data_dir_generator import create_data_dir
+from monkey_island.cc.server_utils.consts import (
+    DEFAULT_DATA_DIR,
+    DEFAULT_DEVELOP_SERVER_CONFIG_PATH,
+    DEFAULT_SERVER_CONFIG_PATH,
+)
 
 
-def create_default_config_file(path):
+def create_default_server_config_file() -> str:
+    if not os.path.isfile(DEFAULT_SERVER_CONFIG_PATH):
+        create_data_dir(DEFAULT_DATA_DIR, False)
```

Comment:
I'm sorry, but I still don't understand what's wrong with using `os.makedirs` here. The explanation of `$HOME` might not be resolved doesn't make sense. It's the responsibility of `DEFAULT_DATA_DIR` to point to a resolvable path. And if `DEFAULT_DATA_DIR` can't be resolved, how will not making parent directories help us?

Reply:
For example, on Linux, if the $HOME env variable doesn't exist, the value of `DEFAULT_DATA_DIR` will remain "$HOME/.monkey_island" (we're using `os.path.expandvars()` to resolve the path but in case nothing is resolved, it will simply return the same string, it won't throw any kind of error), and using `os.makedirs` will create a directory named "$HOME" and the ".monkey_island" directory under that. Creating a directory named "$HOME" is not the expected behaviour. If the $HOME variable can't be resolved, we expect the user to specify the path for the data directory.

Reply:
> If the $HOME variable can't be resolved, we expect the user to specify the path for the data directory.

 How is this expressed in the code and where? Users and developers are expected to get `os.mkdir` error and know that `os.mkdir` failed because it doesn't create parent folders, so it means that the parent folder of `data_dir` is not present? 
What prevents the user from making the same mistake via cmd arguments? 
Wouldn't it be better to 
```
if not os.path.isdir(data_dir_parent):
    logger.error(f"Can't create a data directory because path {data_dir_parent} doesn't exist. Specify a reachable directory with {data_dir_flag}")
```

Reply:
See [baee74b7](https://github.com/guardicore/monkey/pull/1170/commits/baee74b761b2d343688ba898f161c52215b47c10)